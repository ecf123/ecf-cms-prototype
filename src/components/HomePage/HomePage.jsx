import "./HomePage.scss";
import Button from "../../components/Button/Button";
import logo from "../../assets/ECFP-company-logo.svg";

const HomePage = () => {
  return (
    <div className="homePage">
      <div className="homePage__container">
        <div>
          <div className="homePage__logo">
            <div className="homePage__tag">
              <img src={logo} alt="ECF-CMS logo" className="homePage__image" />
              <h2 className="homePage__tagText">
                <span className="homePage__span">THE</span>
                <br />
                FUTURE HUB
              </h2>
            </div>
          </div>
        </div>
        <h2 className="homePage__heading">
          Welcome to the<br></br> ECF - CMS
        </h2>
        <p className="homePage__tagline">
          The Early Careers Foundation Content Management System
        </p>
        <div className="homePage__buttons">
          <Button
            style={"button login-black rounded-square-border modal"}
            textStyle={"text medium-text black"}
            buttonText={"Create An Account"}
          />
        </div>
        <Button
          style={"button light-grey rounded-square-border modal"}
          textStyle={"text medium-text light-grey"}
          buttonText={"Cancel"}
        />
      </div>
    </div>
  );
};

export default HomePage;
