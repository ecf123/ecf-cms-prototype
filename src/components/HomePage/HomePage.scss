@use "../../styles/variables.scss" as *;
@use "../../styles/typography.scss" as *;

.homePage {
  &__heading {
    color: $color-blue-navy-dark;
    font-weight: $font-weight-regular;
  }

  &__image {
    width: 5rem;
  }
  &__logo {
    display: flex;
    margin-bottom: 4rem;
    margin-right: 1rem;
  }
  &__tag {
    display: flex;
    text-align: left;
    margin-top: 4rem;
  }
  &__tagText {
    margin: auto;
    font-size: $font-size-medium;
    color: $color-blue-navy-dark;
  }

  &__span {
    color: $color-blue-navy-dark;
    font-weight: $font-weight-regular;
  }
  &__container {
    margin: auto;
    max-width: 25rem;
    height: 50rem;
    display: flex;
    align-items: center;
    text-align: center;
    flex-direction: column;
    justify-content: center;
  }

  &__tag {
    color: $color-grey-light;
  }

  &__tagline {
    margin-top: 2rem;
    width: 20rem;
    color: $color-white-blue;
  }

  &__buttons {
    margin: 1rem;
  }
}
