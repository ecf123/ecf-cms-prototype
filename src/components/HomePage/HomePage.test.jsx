import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import HomePage from "./HomePage";

it("Should render the home page", () => {
    const { container } = customRender(<HomePage />);
    expect(container).toMatchSnapshot();
  });