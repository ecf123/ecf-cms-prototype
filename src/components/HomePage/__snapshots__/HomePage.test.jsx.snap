// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the home page 1`] = `
<div>
  <div
    class="homePage"
  >
    <div
      class="homePage__container"
    >
      <div>
        <div
          class="homePage__logo"
        >
          <div
            class="homePage__tag"
          >
            <img
              alt="ECF-CMS logo"
              class="homePage__image"
              src="ECFP-company-logo.svg"
            />
            <h2
              class="homePage__tagText"
            >
              <span
                class="homePage__span"
              >
                THE
              </span>
              <br />
              FUTURE HUB
            </h2>
          </div>
        </div>
      </div>
      <h2
        class="homePage__heading"
      >
        Welcome to the
        <br />
         ECF - CMS
      </h2>
      <p
        class="homePage__tagline"
      >
        The Early Careers Foundation Content Management System
      </p>
      <div
        class="homePage__buttons"
      >
        <button
          class="button login-black rounded-square-border modal"
        >
          <div />
          <p
            class="text medium-text black"
          >
            Create An Account
          </p>
        </button>
      </div>
      <button
        class="button light-grey rounded-square-border modal"
      >
        <div />
        <p
          class="text medium-text light-grey"
        >
          Cancel
        </p>
      </button>
    </div>
  </div>
</div>
`;
